packages/plugin-version/sources/commands/version/apply.ts

```diff
@@ -0,0 +1,147 @@
+import {WorkspaceRequiredError}                                                     from '@berry/cli';
+import {AllDependencies, Cache, Configuration, IdentHash, LightReport, Manifest}    from '@berry/core';
+import {MessageName, PluginConfiguration, Project, StreamReport, WorkspaceResolver} from '@berry/core';
+import {Workspace, structUtils}                                                     from '@berry/core';
+import {PortablePath}                                                               from '@berry/fslib';
+import {Writable}                                                                   from 'stream';
+
+import * as versionUtils                                                            from '../../versionUtils';
+
+const SUPPORTED_UPGRADE_REGEXP = /^([~^]?)[0-9]+\.[0-9]+\.[0-9]+$/;
```

Comment:
so pre-release versions aren't supported?

Reply:
Ideally it should be `[~^]{{whatever semver allows}}` 🤔 I'll try to use a more complex regexp.

Ref https://github.com/semver/semver/issues/232

Reply:
Ideally it should be `[~^]{{whatever semver allows}}` 🤔 I'll try to use a more complex regexp.

Ref https://github.com/semver/semver/issues/232